edx_exams/apps/core/api.py

```diff
@@ -0,0 +1,80 @@
+"""
+Library for the edx-exams service.
+"""
+import logging
+from datetime import datetime
+
+import pytz
+
+from edx_exams.apps.core.models import ExamAttempt
+from edx_exams.apps.core.statuses import ExamAttemptStatus
+
+log = logging.getLogger(__name__)
+
+
+def update_attempt_status(attempt_id, to_status):
+    """
+    Helper method to handle state transition of attempt status
+    """
+
+    attempt_obj = ExamAttempt.get_attempt_by_id(attempt_id)
+
+    exam_id = attempt_obj.exam.id
+    user_id = attempt_obj.user.id
+
+    from_status = attempt_obj.status
+
+    log.info(
+        'Updating attempt status for exam_id=%(exam_id)s for user_id=%(user_id)s '
+        'from status "%(from_status)s" to "%(to_status)s"',
+        {
+            'exam_id': exam_id,
+            'user_id': user_id,
+            'from_status': from_status,
+            'to_status': to_status,
+        }
+    )
+
+    attempt_obj.status = to_status
+
+    if to_status == ExamAttemptStatus.started:
+        attempt_obj.start_time = datetime.now(pytz.UTC)
```

Comment:
The allowed time is also calculated here, working on that now

Reply:
related, I think we should now move this into your core/api file https://github.com/edx/edx-exams/blob/main/edx_exams/apps/api/utils.py#L12